### PR TITLE
Add document for -mstrict-align and -mno-strict-align

### PR DESCRIPTION
**Pull request by kito-cheng**
We have those two option for a while, but not document in this repo yet, however we are intend to claify the scope of the option recently, so it's time to document down what we have agree among different community.

### CODE REVIEW

**Comment by kito-cheng** [comment]
Ref: 
Discussion in riscv-c-api-doc: https://github.com/riscv-non-isa/riscv-c-api-doc/issues/73
GCC patch: https://patchwork.sourceware.org/project/gcc/patch/cc653588-ac19-4f09-9693-78a4d47cfb1f@gmail.com/

**Comment by kito-cheng** [comment]
CC: @nick-knight @kito-cheng @vineetgarc @asb @aswaterman

**Comment by topperc** [comment]
CC @preames

File: README.mkd [hunk 1]
```diff
@@ -400,11 +400,22 @@ specification, and in the draft [CORE-V Builtin
...
 specification.
 
+## Common Toolchain Command Line Options
+
+This section lists common RISC-V specific toolchain command line options.
+
+### `-mstrict-align`/`-mno-strict-align`
+Control the code generation for scalar memory access with strict alignment (also known as unaligned access or misaligned access).
+
+`-mstrict-align`: The compiler disallows misaligned access for scalar memory access.
```
Comment (topperc) [inline README.mkd:410]: Borrowing from AArch64 gcc documentation. Perhaps

```
indicates that the compiler should not assume that unaligned scalar memory references are handled by the system.
```

File: README.mkd [hunk 2]
```diff
@@ -400,11 +400,22 @@ specification, and in the draft [CORE-V Builtin
 Function](https://github.com/openhwgroup/core-v-sw/blob/master/specifications/corev-builtin-spec.md)
 specification.
 
+## Common Toolchain Command Line Options
+
+This section lists common RISC-V specific toolchain command line options.
+
+### `-mstrict-align`/`-mno-strict-align`
+Control the code generation for scalar memory access with strict alignment (also known as unaligned access or misaligned access).
```
Comment (nick-knight) [inline README.mkd:408]:
```suggestion
Control the code generation for scalar memory access with strict alignment (non-strict also known as unaligned access or misaligned access).
```

**Review by nick-knight — APPROVED** [review]
I have no objection to documenting the current behavior.

**Review by cmuellner — APPROVED** [review]
LGTM.
Should we also mention that intrinsics are covered, but inline-assembly is not?

**Comment by kito-cheng** [comment]
Update to reflect the discussion within GCC and LLVM community: keep it control both scalar and vector unaligned access, because LLVM already use that semantic for more than one release, change it is not good idea, also GCC 14 isn't really control vector memory by -mstrict-align...it's always allowed in GCC 14, so we'll gonna "fix" GCC 14 to fix this behavior

**Comment by kito-cheng** [comment]
Update:
- Update the option priority for unaligned access.

**Review by topperc — APPROVED** [review]
LGTM

**Comment by kito-cheng** [comment]
Gonna merge since I believe we reach consensus between LLVM and GCC community

**Pull request merged**